settings.go

```diff
@@ -0,0 +1,80 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	// "errors"
+	"fmt"
+	// "strings"
+
+	"github.com/kubewarden/container-resources-policy/resource"
+	kubewarden "github.com/kubewarden/policy-sdk-go"
+	kubewarden_protocol "github.com/kubewarden/policy-sdk-go/protocol"
+)
+
+type ResourceConfiguration struct {
+	MaxLimit       resource.Quantity `json:"maxLimit"`
+	DefaultRequest resource.Quantity `json:"defaultRequest"`
+	DefaultLimit   resource.Quantity `json:"defaultLimit"`
+}
+
+type Settings struct {
+	Cpu          *ResourceConfiguration `json:"cpu,omitempty"`
+	Memory       *ResourceConfiguration `json:"memory,omitempty"`
+	IgnoreImages []string               `json:"ignoreImages,omitempty"`
+}
+
+func (r *ResourceConfiguration) valid() error {
+	if r.MaxLimit.IsZero() && r.DefaultLimit.IsZero() && r.DefaultRequest.IsZero() {
+		return fmt.Errorf("all the quantities must be defined")
+	}
+	if r.MaxLimit.Cmp(r.DefaultLimit) < 0 ||
+		r.MaxLimit.Cmp(r.DefaultRequest) < 0 {
+		return fmt.Errorf("default values cannot be greater than the max limit")
+	}
+
+	return nil
+}
+
+func (s *Settings) Valid() (bool, error) {
+	if s.Cpu == nil && s.Memory == nil {
+		return false, fmt.Errorf("no settings provided. At least one resource limit or request must be verified")
```

Comment:
I think it would be nice to not exit on the 1st validation failure. let's keep joining them until the end of the method.

The signature of the method can also be changed to return just an `error`, not a `(bool, error)`. The caller can then check if `err == nil`